File: library/src/main/java/com/ltst/chessview/DimenTools.java
package com.ltst.chessview;

import android.content.Context;
import android.graphics.Point;
import android.util.TypedValue;
import android.view.Display;
import android.view.WindowManager;

public class DimenTools {
    public static float dpFromPx(Context context, float px) {
        return px / context.getResources().getDisplayMetrics().density;
    }

    public static float pxFromDp(Context context, float dp) {
        return dp * context.getResources().getDisplayMetrics().density;
    }

    public static float pxByDensity(Context context, int px) {
        return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, px, context.getResources().getDisplayMetrics());
    }

    public static float spFromPx(Context contex, float px) {
        return px / contex.getResources().getDisplayMetrics().scaledDensity;
    }

    public static float pxFromSp(Context contex, float sp) {
        return sp * contex.getResources().getDisplayMetrics().scaledDensity;
    }

    public static Point displaySize(Context context) {
        WindowManager wm = (WindowManager) context.getSystemService(Context.WINDOW_SERVICE);
        Display display = wm.getDefaultDisplay();
        Point size = new Point();
        display.getSize(size);
        return size;
    }

}

File: library/src/main/java/com/cosic/chessview/utils/DimenTools.kt
package com.cosic.chessview.utils

import android.content.Context
import android.graphics.Point
import android.util.TypedValue
import android.view.WindowManager

object DimenTools {

    fun dpFromPx(context: Context, px: Float): Float {
        return px / context.resources.displayMetrics.density
    }

    @JvmStatic
    fun pxFromDp(context: Context, dp: Float): Float {
        return dp * context.resources.displayMetrics.density
    }

    fun pxByDensity(context: Context, px: Int): Float {
        return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, px.toFloat(), context.resources.displayMetrics)
    }

    fun spFromPx(context: Context, px: Float): Float {
        return px / context.resources.displayMetrics.scaledDensity
    }

    @JvmStatic
    fun pxFromSp(context: Context, sp: Float): Float {
        return sp * context.resources.displayMetrics.scaledDensity
    }

    fun displaySize(context: Context): Point {
        val wm = context.getSystemService(Context.WINDOW_SERVICE) as WindowManager
        val display = wm.defaultDisplay
        val size = Point()
        display.getSize(size)
        return size
    }
}